libvast/vast/concept/printable/vast/data.hpp

```diff
@@ -54,13 +56,37 @@ struct data_printer : printer<data_printer> {
   }
 };
 
+using json_data_printer = json_printer<policy::tree, 2>;
```

Comment:
I didn't delete `vast/concept/printable/vast/json.hpp` header and reuse its printer as a second option for `vast::data` printer:
```
auto text_str = to_string(data_object); // default to_string(data) rendering
auto json_str = to_string(data_object, print_rendering::json); // explicitly set json rendering
```

If the idea of such extended `to_string(data, rendering)` is ok, then `json_data_printer` can be moved here entirely.

Reply:
For `YAML` we use this:

```
/// Prints data as YAML.
/// @param x The data instance.
/// @returns The YAML representation of *x*, or an error.
caf::expected<std::string> to_yaml(const data& x);

/// Parses YAML into a data.
/// @param str The string containing the YAML content
/// @returns The parsed YAML as data, or an error.
caf::expected<data> from_yaml(std::string_view str);
```

So I think you can just name the function `to_json` and have it return a `caf::expected<std::string>`.